controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1099,12 +1254,51 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
                 });
     }
 
-    private void checkState(State currState, State expectedState) {
-        if (currState != expectedState) {
-            throw StoreException.create(StoreException.Type.ILLEGAL_STATE,
-                    "Stream: " + getName() + " Current State: " + currState.name() + " Expected State:"
-                            + expectedState.name());
-        }
+    @Override
+    public CompletableFuture<Void> createCommittingTransactionsRecord(final int epoch, final List<UUID> txnsToCommit) {
+        return createCommittingTxnRecord(new CommittingTransactionsRecord(epoch, txnsToCommit).toByteArray());
+    }
+
+    @Override
+    public CompletableFuture<CommittingTransactionsRecord> getCommittingTransactionsRecord() {
+        CompletableFuture<CommittingTransactionsRecord> result = new CompletableFuture<>();
+        getCommittingTxnRecord()
```

Comment:
We seem to be using `Txn` and `Transaction` inconsistently in method names. I'm fully fine with `Txn` for singular and `Txns`  for plural, but I'd rather be consistent in the method names if possible.

Reply:
created a new issue https://github.com/pravega/pravega/issues/2682